components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java

```diff
@@ -458,7 +460,14 @@ protected JWTClaimsSet createJWTClaimSet(OAuthAuthzReqMessageContext authAuthzRe
         if (StringUtils.isNotEmpty(scope)) {
             jwtClaimsSetBuilder.claim(SCOPE, scope);
         }
-
+        if (OAuth2ServiceComponentHolder.isAcrColumnEnabled()) {
+            if (isAuthCodeGrant(tokenReqMessageContext)) {
+                Optional<String> selectedACRValueOptional = OAuth2Util.getSelectedACRValue(tokenReqMessageContext);
+                selectedACRValueOptional.ifPresent(acrValue -> jwtClaimsSetBuilder.claim(OAuth2Util.ACR, acrValue));
+            } else if (isRefreshTokenGrant(tokenReqMessageContext)) {
+                jwtClaimsSetBuilder.claim(OAuth2Util.ACR, tokenReqMessageContext.getAcr());
```

Comment:
Update the logic

Reply:
As mentioned [here](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1984#discussion_r1064569753) check if we can get the ACR value to the refresh token in some other way. If possible we need to update this code snippet